File: frontend/src/types/course.ts
export interface CourseHole {
  hole_number: number;
  par: number;
  distance_meters: number;
}

export interface Course {
  id: number;
  name: string;
  region: string;
  address: string | null;
  description: string | null;
  holes_count: number;
  par: number;
  difficulty: string;
  green_fee_avg: number | null;
  tags: string | null;
}

export interface CourseDetail extends Course {
  holes: CourseHole[];
}

File: backend/app/schemas/course.py
"""Course/CourseHole 응답 스키마."""
from pydantic import BaseModel, ConfigDict


class CourseHoleRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    hole_number: int
    par: int
    distance_meters: int


class CourseRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    region: str
    address: str | None = None
    description: str | None = None
    holes_count: int
    par: int
    difficulty: str
    green_fee_avg: int | None = None
    tags: str | None = None


class CourseDetailRead(CourseRead):
    holes: list[CourseHoleRead] = []
